derive.cc

```diff
@@ -0,0 +1,24 @@
+#include "derive.h"
+
+using namespace std;
+int main() {
+	Func f('x');
+	Func x('x');
+	//Func g(5.);
+	Func h = f * 5.;
+	Func const1(23.);
+	Func const2(2.);
+	Func mult_consts = const1 * const2;
+	cout << mult_consts << endl;
+	cout << "h=" << h << "derivative: " << h.derivative() << endl;
+	Func long_func = 2. * h + 5. * pow((f + 20.), 2.);
+	cout << "long_func = " << long_func << "derivative: " << long_func.derivative() << endl;
+	Func g = f*0.;
+	cout << "g(x)=" << g << " g'(x) " << g.derivative() << endl;	
+	Func f_mult_g = f * g;
+	cout << "h is: " << h << " derivative is: " << h.derivative() << endl;
+	cout << "f_mult_g is: " << f_mult_g << " derivative is: " << f_mult_g.derivative() << endl;
+	Func sub_func = f * 3. - x;
+	cout << "sub_func(x) = " << sub_func << " sub_func'(x) = " << sub_func.derivative() << endl;
```

Comment:
Could you please add tests with zero coeff and very small coeff.

Reply:
I've got a test with zero coeff - line 18. I  will add now another test - more elaborate and a small coeff as well.

Reply:
Added another zero coeff and small coef.